apps/aeutils/src/aeu_mtrees.erl

```diff
@@ -252,3 +259,60 @@ proof_db_commit(_Cache,_DB) ->
 
 proof_db_fold(Fun, Initial, Proof) ->
     dict:fold(Fun, Initial, Proof).
+
+%%%===================================================================
+%%% serialization
+%%%===================================================================
+-spec serialize(mtree()) -> binary().
+serialize(Tree) ->
+    LeavesBins =
+        lists:map(
+            fun({Key, Value}) ->
+                aec_object_serialization:serialize(
+                    mtree_leaf,
+                    ?VSN,
+                    leaf_serialization_template(?VSN),
+                    [ {key, Key}
+                    , {val, Value}
+                    ])
+            end,
+            to_list(Tree)),
```

Comment:
This should be done with an iterator instead. We don't want to construct the intermediate list.